removed deprecated NotebookSpec files

kannanjayachandran opened the pull request:
### Description

Issue #12307
Remove NotebookSpec from conda env

### Checklist - did you ...

- [x] (I added a descriptive title)
- [x] (I searched open requests and couldn't find a duplicate)

Removed the following files;

- [`__init__.py` file](https://github.com/conda/conda/blob/main/conda_env/specs/__init__.py)
- [`notebook.py` file](https://github.com/conda/conda/blob/main/conda_env/specs/notebook.py)
- [Related test files](https://github.com/conda/conda/blob/main/tests/conda_env/specs/test_notebook.py)

beeankha commented:
@conda-bot check

kannanjayachandran commented:
> Hi @kannanjayachandran,
> 
> Thanks for trying to help us out! Unfortunately the tests are failing 😭. You won't be able to just delete the [**init**.py](https://github.com/conda/conda/blob/main/conda_env/specs/__init__.py) file that I referenced in the original issue. You have to carefully remove the usage of `NotebookSpec` from it.
> 
> We still need that module for handling the other file types.
> 
> Let me know if that makes sense.

oh I totally got it wrong I will make an updated PR

beeankha commented:
> what about the other two files, - [notebook.py file](https://github.com/conda/conda/blob/main/conda_env/specs/notebook.py), [Related test files](https://github.com/conda/conda/blob/main/tests/conda_env/specs/test_notebook.py) can I remove them (delete file) or update them

You should be able to just delete those two files completely. 

Additionally, it would be preferable for you to update this PR vs create a new one, if possible!

kannanjayachandran commented:
> > what about the other two files, - [notebook.py file](https://github.com/conda/conda/blob/main/conda_env/specs/notebook.py), [Related test files](https://github.com/conda/conda/blob/main/tests/conda_env/specs/test_notebook.py) can I remove them (delete file) or update them
> 
> You should be able to just delete those two files completely.
> 
> Additionally, it would be preferable for you to update this PR vs create a new one, if possible!

Ok got it. Yes I would be updating this same PR instead of creating a new one

kannanjayachandran commented:
Hi @beeankha  I tried to modify this same PR but it was giving me errors as the repository had recent commits, hence I had to make another [Pull request](https://github.com/conda/conda/pull/12363) hope that's fine. 

I have updated the  [`__init__.py` file](https://github.com/conda/conda/blob/main/conda_env/specs/__init__.py) file and removed  [`notebook.py` file](https://github.com/conda/conda/blob/main/conda_env/specs/notebook.py),  [Related test files](https://github.com/conda/conda/blob/main/tests/conda_env/specs/test_notebook.py).

one more thing,  should I close this PR
